src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -317,12 +317,27 @@ def purge(self, service_name: str) -> None:
             '--namespace', cast(str, spec.service_id),
             'rm', 'grace',
         ]
-        subprocess.run(
-            cmd,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            timeout=10
-        )
+        try:
+            result = subprocess.run(
```

Comment:
This is somewhat orthogonal to the actual change being made, but why does this shell out to the `rados` command line tool? Other parts of the mgr are able to use ceph rados api calls.
(Especially since this class has a `self.mgr` attribute available!)